libpromises/generic_agent.c

```diff
@@ -348,6 +392,130 @@ static bool LoadAugmentsData(EvalContext *ctx, const char *filename, const JsonE
             JsonDestroy(vars);
         }
 
+        /* Uses the new format allowing metadata (CFE-3633) */
+        element = JsonObjectGet(augment, "variables");
+        if (element != NULL)
+        {
+            JsonElement* variables = JsonExpandElement(ctx, element);
+
+            if (variables == NULL || JsonGetType(variables) != JSON_TYPE_OBJECT)
+            {
+                Log(LOG_LEVEL_ERR, "Invalid augments variables in '%s', must be a JSON object", filename);
+                goto variables_cleanup;
+            }
+
+            JsonIterator variables_iter = JsonIteratorInit(variables);
+            const char *vkey;
+            while ((vkey = JsonIteratorNextKey(&variables_iter)))
```

Comment:
Don't include assignments inside expressions, unless they avoid great repetition & use explicit comparisons.
 - [CONTIBUTING.md](https://github.com/cfengine/core/blob/master/CONTRIBUTING.md)

Reply:
This is just copy-pasted from the nearby code, I'm not changing it there. Besides, I think iterations like this should be an exception.